### PR TITLE
Add document describing development practices

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: dev-docs/development-process.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+You may need to provde additional configuration, either as environment variables or in `user-config.yml`.
+You can find a template for `user-config.yml` in `user-config-example.yml`: just copy the latter to the former and edit it.
+
+## Hacking on Clients
+
+The clients are in `clients/`.
+One, `client-web`, is designed for use in web browsers, while the other, `client`, is intended for use in Node environments.
+In fact, it is used by all of the services and libraries in the repository.
+Note that taskcluster-ui does not use `client-web`, since it communicates with taskcluster-web-server via GraphQL.
```
Comment (helfi92) [inline dev-docs/development-process.md]: Nit: `web-server` instead of taskcluster-web-server

**Comment by helfi92** [comment]
Tests are failing, otherwise LGTM :)

**Comment by djmitche** [comment]
Lol, I even wrote about running `yarn generate`..

**Comment by helfi92** [comment]
Haha

**Pull request merged**